codewind-che-sidecar/tests/sidecarfvt.bats

```diff
@@ -0,0 +1,206 @@
+#!/usr/bin/env bats
+
+#*******************************************************************************
+# Copyright (c) 2019 IBM Corporation and others.
+# All rights reserved. This program and the accompanying materials
+# are made available under the terms of the Eclipse Public License v2.0
+# which accompanies this distribution, and is available at
+# http://www.eclipse.org/legal/epl-v20.html
+#
+# Contributors:
+#     IBM Corporation - initial API and implementation
+#*******************************************************************************
+
+# Load common utility functions
+load testutil
+
+# Setup code to run before each test
+setup() {
+    export time_before=$SECONDS
+
+    if [ -z "$CLUSTER_IP" ]; then
+        echo "# Cluster IP is not defined in " '$CLUSTER_IP' >&3
+        exit 1
+    fi
+
+    if [ -z "$CHE_NAMESPACE" ]; then
+        echo "# Che namespace is not defined in " '$CHE_NAMESPACE' >&3
+        exit 1
+    fi
+
+    export CODEWIND_DEVFILE_URL=https://raw.githubusercontent.com/eclipse/codewind-che-plugin/master/devfiles/0.2.0/devfile.yaml
+    export CODEWIND_DEVILE_JSON=devfile.json
+    export CHE_INGRESS_DOMAIN=http://che-$CHE_NAMESPACE.$CLUSTER_IP.nip.io
+    export KUBE_NAMESPACE_ARG="-n $CHE_NAMESPACE"
+
+    # Discover workspace ID written into temporary file during workspace creation
+    if [ -f che_workspace_id.txt ]; then
+        export CHE_WORKSPACE_ID=$(cat che_workspace_id.txt)
+    else
+        export CHE_WORKSPACE_ID=workspace00000
+    fi
+
+    # Discover workspace pod and sidecar full names based on workspace ID
+    export CHE_WORKSPACE_POD_FULLNAME=$(kubectl get pods -l che.original_name=che-workspace-pod --no-headers -o custom-columns=":metadata.name" $KUBE_NAMESPACE_ARG | grep $CHE_WORKSPACE_ID)
+    export SIDECAR_CONTAINER_FULLNAME=$(kubectl get pods $CHE_WORKSPACE_POD_FULLNAME -o jsonpath='{.spec.containers[*].name}' $KUBE_NAMESPACE_ARG | sed 's/ /\n/g' | grep ^codewind-che-sidecar)
+}
+
+# Teardown code after each test
+teardown() {
+    time_after=$SECONDS
+    echo "# Test time taken: " $(($time_after-$time_before)) " seconds" >&3
+}
+
+@test "Codewind Sidecar Test #1: Create Che workspace from Codewind dev file" {
+    # TODO: convert live .yaml dev file in url to local json file?
+    # TODO: always assume just one codewind related workspace? (=> no need to store workspace ID), 
+    #       if not, use global var outside of bats to store workspace ID rather than local file?
+    # TODO: use dynamically generated workspace name?, delete existing workspace if same name?
+
+    createCodewindCheWorkspace
+}
+
+@test "Codewind Sidecar Test #2: Verify Codewind workspace pod is running" {
+    # Check if pod has started, timeout after 5 minutes
+    runtime="5 minute"
+    endtime=$(date -ud "$runtime" +%s)
```

Comment:
I get an error at this line in the macOS bash when I try to run the scripts:
```
johns-mbp-3:~ johncollier$ runtime="5 minute"
johns-mbp-3:~ johncollier$ date -ud "$runtime" +%s
usage: date [-jnRu] [-d dst] [-r seconds] [-t west] [-v[+|-]val[ymwdHMS]] ... 
            [-f fmt date | [[[mm]dd]HH]MM[[cc]yy][.ss]] [+format]
```

Looks like options that are only available on Linux. We might want to change these lines so that the tests can be run on macOS easily.

Reply:
Ok will try to do something that works across Mac/Linux/other platforms